### ISSUE TITLE
Error when clicking Ctrl + S

### ISSUE THREAD

**Issue by sina-Kim** [spyder-ide/spyder#16422]
### What steps will reproduce the problem?

1. Write the code
2. I habitually clicked Ctrl + S
3. Sometimes the Issue Reporter window opens in this situation
4. Expected to occur during autopep8 operation.

### Traceback
```python-traceback
Traceback (most recent call last):
  File "C:\Users\sinaKim\miniconda3\lib\site-packages\spyder\plugins\editor\widgets\editor.py", line 1904, in _save_file
    self.modification_changed(index=index)
  File "C:\Users\sinaKim\miniconda3\lib\site-packages\spyder\plugins\editor\widgets\editor.py", line 2360, in modification_changed
    finfo = self.data[index]
IndexError: list index out of range
```

## Versions

* Spyder version: 5.1.4 
* Python version: 3.9.6
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.27.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.0 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.2.0 (OK)
nbconvert >=4.0               :  6.1.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.2 (OK)
parso >=0.7.0;<0.9.0          :  0.8.2 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.10.0 (OK)
pylint >=2.5.0;<2.10.0        :  2.9.6 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.2;<1.3.0          :  1.2.2 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.1.0             :  5.1.1 (OK)
qtpy >=1.5.0                  :  1.11.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                :  4.2.0 (OK)
spyder_kernels >=2.1.1;<2.2.0 :  2.1.1 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.5 (OK)
zmq >=17                      :  22.2.1 (OK)

# Optional:
cython >=0.21                 :  None (OK)
matplotlib >=2.0.0            :  3.4.3 (OK)
numpy >=1.7                   :  1.19.5 (OK)
pandas >=1.1.1                :  1.3.2 (OK)
scipy >=0.17.0                :  1.7.1 (OK)
sympy >=0.7.3                 :  None (OK)
```

**Comment by steff456** [comment]
Hi @sina-Kim,

How occasionally do you type Ctrl + S in the editor? It will be great if you find a way to find how to reproduce this issue so we can try it on our end.

Thanks for reporting!

**Comment by sina-Kim** [comment]
Hello. @steff456  

After writing a few lines of code, I habitually press Ctrl + S . (Sometimes, press 2-3 times in a row.)

Recently, I came to know the View > Panes > Outline function and organized the Cell as shown below.
```
# %% (1) bla bla
empty cell
# %% (2) bla bla
some codes
# %% (3) bla bla
empty cell
# %% (4) bla bla
```

And hit Ctrl + S. I had the same error as the issue, and I've had the same error ever since.

**Comment by steff456** [comment]
Did you do the Ctrl + S while being in focus in the Outline explorer or the Editor?

**Comment by ccordoba12** [comment]
This is fixed in our latest version (5.3.3), so please update.
